include/boost/url/detail/optional_string.hpp

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright (c) 2019 Vinnie Falco (vinnie.falco@gmail.com)
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_DETAIL_OPTIONAL_STRING_HPP
+#define BOOST_URL_DETAIL_OPTIONAL_STRING_HPP
+
+#include <boost/url/detail/config.hpp>
```

Comment:
you dont need to include config in detail files (since they are never included first)